lib/ansible/module_utils/junos.py

```diff
@@ -60,16 +69,106 @@ def connect(self, **kwargs):
             msg = 'failed to connecto to %s:%s - %s' % (host, port, str(exc))
             self.module.fail_json(msg=msg)
 
-    def send(self, commands):
+        if self.shell._matched_prompt.strip().endswith('%'):
+            self.send('cli')
+        self.send('set cli screen-length 0')
+
+    def send(self, commands, **kwargs):
         return self.shell.send(commands)
```

Comment:
Should this inherit the

    except ShellError:
        module.fail_json(msg='failed to run commands')

from junos_command, instead of losing the try block entirely?